scapy/arch/pcapdnet.py

```diff
@@ -324,7 +325,12 @@ def recv(self, x = MTU):
           else:
             return
       def send(self, x):
-          cls = conf.l2types[1]
+          ll = self.ins.datalink()
+          if ll in conf.l2types:
+              cls = conf.l2types[ll]
+          else:
+              cls = conf.default_l2
+              warning("Unable to guess datalink type (interface=%s linktype=%i). Using %s" % (self.iface, ll, cls.name))
```

Comment:
Make `send` detects when it should add `Loopback()/` instead of `Ether()/`. Already existed on other implementations.